Location dependent formatting in `LocationFormatter`

The number format of the `LocationFormatter` depends on the current Java local.
```
java.lang.AssertionError: expected [24°59'15.486"N] but found [24°59'15,486"N]
Expected :24°59'15.486"N
Actual   :24°59'15,486"N
``` 
Make the number format independet on the current local.

--- a/jpx/src/main/java/io/jenetics/jpx/format/LocationFormatter.java
+++ b/jpx/src/main/java/io/jenetics/jpx/format/LocationFormatter.java
@@ -22,15 +22,9 @@ package io.jenetics.jpx.format;
 import static java.util.Objects.requireNonNull;
 
 import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
 import java.text.NumberFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
@@ -710,8 +704,11 @@ public final class LocationFormatter {
 								}
 								fmt.add(LocationFieldFormat.of(
 									field.get(),
-									() -> new DecimalFormat(
-										field.get().toDecimalPattern(token))
+									() -> {
+										DecimalFormatSymbols symbols = DecimalFormatSymbols.getInstance(Locale.US);
+										return new DecimalFormat(
+										field.get().toDecimalPattern(token), symbols);
+									}
 								));
 							} else {
 								fmt.add(ConstFormat.of(token));

--- a/jpx/src/main/java/io/jenetics/jpx/format/LocationFormatter.java
+++ b/jpx/src/main/java/io/jenetics/jpx/format/LocationFormatter.java
@@ -510,7 +510,10 @@ public final class LocationFormatter {
 		 * @throws NullPointerException if one of the arguments is {@code null}
 		 */
 		public Builder append(final Location.Field field, final String pattern) {
-			return append(field, () -> new DecimalFormat(pattern));
+			return append(field, () -> {
+				DecimalFormatSymbols symbols = DecimalFormatSymbols.getInstance(Locale.US);
+				return new DecimalFormat(pattern, symbols);
+			});
 		}
 
 		/**

--- a/jpx/src/test/java/io/jenetics/jpx/format/LocationFormatterTest.java
+++ b/jpx/src/test/java/io/jenetics/jpx/format/LocationFormatterTest.java
@@ -206,5 +206,16 @@ public class LocationFormatterTest {
 		Assert.assertEquals(locationFormatter.format(location), "23.99");
 	}
 
-
+	@Test
+	public void testParsePattern() {
+		Locale.setDefault(Locale.GERMANY);
+		DecimalFormatSymbols instance = DecimalFormatSymbols.getInstance();
+		Location location = Location.of(Latitude.ofDegrees(23.987635));
+		LocationFormatter locationFormatter = LocationFormatter.builder()
+			.appendPattern("DD.DD")
+			.build();
+		Assert.assertEquals(instance.getDecimalSeparator(), ',');
+		Assert.assertEquals(locationFormatter.toPattern(), "DD.DD");
+		Assert.assertEquals(locationFormatter.format(location), "23.99");
+	}
 }

--- a/jpx/src/test/java/io/jenetics/jpx/format/LocationFormatterTest.java
+++ b/jpx/src/test/java/io/jenetics/jpx/format/LocationFormatterTest.java
@@ -38,9 +38,12 @@ import static io.jenetics.jpx.format.LocationFormatter.ISO_LON_SHORT;
 import static io.jenetics.jpx.format.LocationFormatter.ISO_MEDIUM;
 import static io.jenetics.jpx.format.LocationFormatter.ISO_SHORT;
 
+import java.text.DecimalFormatSymbols;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
 
+import io.jenetics.jpx.format.Location.Field;
 import org.testng.Assert;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
@@ -189,5 +192,19 @@ public class LocationFormatterTest {
 			.map(f -> new Object[]{f})
 			.toArray(Object[][]::new);
 	}
+
+	@Test
+	public void testAppend() {
+		Locale.setDefault(Locale.GERMANY);
+		DecimalFormatSymbols instance = DecimalFormatSymbols.getInstance();
+		Location location = Location.of(Latitude.ofDegrees(23.987635));
+		LocationFormatter locationFormatter = LocationFormatter.builder()
+			.append(Field.DEGREE_OF_LATITUDE, "00.00")
+			.build();
+		Assert.assertEquals(instance.getDecimalSeparator(), ',');
+		Assert.assertEquals(locationFormatter.toPattern(), "DD.DD");
+		Assert.assertEquals(locationFormatter.format(location), "23.99");
+	}
+
 
 }

